src/cljs/org/broadinstitute/firecloud_ui/common/components.cljs

```diff
@@ -449,6 +449,49 @@
    (fn [{:keys [refs this]}]
      (.addEventListener (@refs "filter-field") "search" #(when (= (.-value (.-currentTarget %)) "") (react/call :apply-filter this))))})
 
+(react/defc AutocompleteFilter
+  {:apply-filter
+   (fn [{:keys [props refs]}]
+     ((:on-filter props) (common/get-text refs "autocomplete-filter-field")))
+   :component-did-mount
+   (fn [{:keys [refs props locals this]}]
+     (.addEventListener (@refs "autocomplete-filter-field") "search" #(when (= (.-value (.-currentTarget %)) "") (react/call :apply-filter this)))
```

Comment:
Maybe bump the `#(when ...` down to the next line, and then maybe the `(react/call ...` too.